gh_pages.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+SOURCE_BRANCH="development"
+TARGET_BRANCH="gh-pages"
+
+# Pull requests and commits to other branches shouldn't try to deploy, just build to verify
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+    echo "No need to deploy."
+    exit 0
+fi
+
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+ember install ember-cli-github-pages
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+rm -rf `bash -c "ls -a | grep -vE '\.gitignore|\.git|node_modules|bower_components|(^[.]{1,2}/?$)'"` && touch .gitkeep
+git add .
+git commit -m "Deploy to GitHub Pages: ${SHA}"
+git checkout $SOURCE_BRANCH
+
+# Build the ember app to Target Branch
+ember $TARGET_BRANCH:commit --message "Deploy to GitHub Pages: ${SHA}"
```

Comment:
Make sure there exists only _one_ commit on the `gh-pages` repo. Every commit should just overwrite it. And you should force push to it.

Reply:
This handled nicely by https://docs.travis-ci.com/user/deployment/pages/. I don't see why you'd need to use `ember-cli-github-pages` for this.

Reply:
This handled nicely by https://docs.travis-ci.com/user/deployment/pages/. I don't see why you'd need to use `ember-cli-github-pages` for this. 

You can just set the correct path at build time, build it and just point travis to deploy the distribution folder to gh-pages